File: frontend/src/api/management.ts
import axios from 'axios'

export type RuleListName = 'blacklist' | 'whitelist'
export type RuleTarget = 'job_name' | 'company_name' | 'job_description' | 'job_tags'
export type RuleMatchType = 'contains' | 'exact'

export interface JobRule {
  id: string
  keyword: string
  target: RuleTarget
  match_type: RuleMatchType
  enabled: boolean
}

export interface RuleInput {
  keyword: string
  target: RuleTarget
  match_type: RuleMatchType
  enabled: boolean
}

export interface ManagementConfig {
  version: number
  settings: { match_threshold: number }
  blacklist: JobRule[]
  whitelist: JobRule[]
}

export interface RuleTestInput {
  job_name: string
  company_name: string
  job_description: string
}

export interface RuleTestResult {
  result: RuleListName | 'unmatched'
  matched_rule: JobRule | null
}

const client = axios.create({
  baseURL: '/api/v1/management',
  timeout: 15_000,
})

export async function fetchManagementConfig(): Promise<ManagementConfig> {
  const { data } = await client.get<ManagementConfig>('/config')
  return data
}

export async function saveManagementSettings(matchThreshold: number) {
  const { data } = await client.patch<{ match_threshold: number }>('/settings', {
    match_threshold: matchThreshold,
  })
  return data
}

export async function createRule(listName: RuleListName, input: RuleInput) {
  const { data } = await client.post<JobRule>(`/${listName}`, input)
  return data
}

export async function updateRule(
  listName: RuleListName,
  ruleId: string,
  input: Partial<RuleInput>,
) {
  const { data } = await client.patch<JobRule>(`/${listName}/${ruleId}`, input)
  return data
}

export async function deleteRule(listName: RuleListName, ruleId: string) {
  await client.delete(`/${listName}/${ruleId}`)
}

export async function testRules(input: RuleTestInput) {
  const { data } = await client.post<RuleTestResult>('/test', input)
  return data
}

File: backend/app/schemas/management.py
from typing import List, Literal, Optional

from pydantic import BaseModel, Field, constr


RuleTarget = Literal["job_name", "company_name", "job_description", "job_tags"]
# exact 仅用于兼容项目现有的精确岗位名规则；管理页新增规则仍固定为 contains。
RuleMatchType = Literal["contains", "exact"]
RuleListName = Literal["blacklist", "whitelist"]
RuleKeyword = constr(strip_whitespace=True, min_length=1, max_length=120)


class RuleBase(BaseModel):
    keyword: RuleKeyword
    target: RuleTarget
    match_type: RuleMatchType = "contains"
    enabled: bool = True

class RuleCreate(RuleBase):
    pass


class RuleUpdate(BaseModel):
    keyword: Optional[RuleKeyword] = None
    target: Optional[RuleTarget] = None
    match_type: Optional[RuleMatchType] = None
    enabled: Optional[bool] = None

class Rule(RuleBase):
    id: str = Field(min_length=1, max_length=100)


class ManagementSettings(BaseModel):
    match_threshold: int = Field(ge=0, le=100)


class ManagementConfig(BaseModel):
    version: int = 1
    settings: ManagementSettings
    blacklist: List[Rule] = Field(default_factory=list)
    whitelist: List[Rule] = Field(default_factory=list)


class RuleTestPayload(BaseModel):
    job_name: str = ""
    company_name: str = ""
    job_description: str = ""


class RuleMatch(BaseModel):
    list_name: RuleListName
    rule: Rule


class RuleTestResponse(BaseModel):
    result: Literal["blacklist", "whitelist", "unmatched"]
    matched_rule: Optional[Rule] = None
